src/cloud/components/Onboarding/FolderPageInviteSection.tsx

```diff
@@ -0,0 +1,184 @@
+import { mdiClose } from '@mdi/js'
+import copy from 'copy-to-clipboard'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import Button from '../../../design/components/atoms/Button'
+import ColoredBlock from '../../../design/components/atoms/ColoredBlock'
+import Flexbox from '../../../design/components/atoms/Flexbox'
+import Form from '../../../design/components/molecules/Form'
+import { FormSelectOption } from '../../../design/components/molecules/Form/atoms/FormSelect'
+import FormRow from '../../../design/components/molecules/Form/templates/FormRow'
+import FormRowItem from '../../../design/components/molecules/Form/templates/FormRowItem'
+import styled from '../../../design/lib/styled'
+import { SerializedOpenInvite } from '../../interfaces/db/openInvite'
+import { SerializedTeam } from '../../interfaces/db/team'
+import { boostHubBaseUrl } from '../../lib/consts'
+import { useI18n } from '../../lib/hooks/useI18n'
+import { lngKeys } from '../../lib/i18n/types'
+import {
+  OpenInvitesProvider,
+  useOpenInvites,
+} from '../../lib/stores/openInvites'
+import { usePage } from '../../lib/stores/pageStore'
+import { useTeamPreferences } from '../../lib/stores/teamPreferences'
+import { getOpenInviteURL, getTeamURL } from '../../lib/utils/patterns'
+
+const InviteSection = ({ team }: { team: SerializedTeam }) => {
+  const { teamPreferences, toggleItem } = useTeamPreferences()
+  const { translate, getRoleLabel } = useI18n()
+  const store = useOpenInvites()
+  const mountedRef = useRef(false)
+  const [selectedInvite, setSelectedInvite] = useState<SerializedOpenInvite>()
+  const [copyButtonLabel, setCopyButtonLabel] = useState<string>(
+    translate(lngKeys.GeneralCopyVerb)
+  )
+
+  const onButtonClick = useCallback(() => {
+    if (selectedInvite == null) {
+      return
+    }
+
+    copy(
+      `${boostHubBaseUrl}${getTeamURL(team)}${getOpenInviteURL(selectedInvite)}`
+    )
+    setCopyButtonLabel(`✓ ${translate(lngKeys.GeneralCopied)}`)
+    setTimeout(() => {
+      setCopyButtonLabel(translate(lngKeys.GeneralCopyVerb))
+    }, 600)
+  }, [selectedInvite, team, translate])
+
+  useEffect(() => {
+    mountedRef.current = true
+    return () => {
+      mountedRef.current = false
+    }
+  }, [])
+
+  useEffect(() => {
+    if (
+      mountedRef.current &&
+      store.state === 'loaded' &&
```

Comment:
we can check store.state before rendering this comopnent. Checking in effect hook doesn't look good.
Also rather than relying on state updates via hook deps, I'd like to implement a proper custom event listener so we can control when to update specifically.